### DISCUSSION TITLE
Multi-module (full-stack) projects

### DISCUSSION THREAD

**Discussion by nalbion** [Pythagora-io/gpt-pilot#60] [Ideas]
Currently, `ask_for_app_type()` is hard-coded to always return "app", but I'd like to see full-stack and multi-module projects supported.

- frontend (and micro front-ends)
- backend (and micro services)
- ios
- android

Further to https://github.com/Pythagora-io/gpt-pilot/discussions/21, the system should be able to detect or suggest (not dictate) the language and frameworks for each module.

- pom.xml -> existing java, read dependencies, issue management etc
- package.json -> existing JavaScript, read dependencies, issue management etc
  - tsconfig.json -> existing TypeScript

**Comment by zvone187** [comment]
Yea, definitely. Many parts in the code are hardcoded like that just because we didn't have time to debug it. But whenever someone gets to this can just uncomment this but also it needs to be propagated throughout all the prompts so that GPT knows what kind of an app it needs to build.

**Comment by lolidkwhattoputinhereokbye** [comment]
hi
